src/main/java/org/kuali/kfs/module/ar/document/web/struts/CashControlForm.java

```diff
@@ -0,0 +1,141 @@
+                if (!cashControlDetail.getReferenceFinancialDocument().getDocumentHeader().hasWorkflowDocument()) {
+                    // populate workflowDocument in documentHeader, if needed
+                    WorkflowDocument workflowDocument = SpringContext.getBean(SessionDocumentService.class)
+                            .getDocumentFromSession(GlobalVariables.getUserSession(),
+                                    cashControlDetail.getReferenceFinancialDocumentNumber());
+
+                    if (workflowDocument == null) {
+                        // gets the workflow document from doc service
+                        workflowDocument = SpringContext.getBean(WorkflowDocumentService.class)
+                                .loadWorkflowDocument(cashControlDetail.getReferenceFinancialDocumentNumber(),
+                                        GlobalVariables.getUserSession().getPerson());
+                        SpringContext.getBean(SessionDocumentService.class)
+                                .addDocumentToUserSession(GlobalVariables.getUserSession(), workflowDocument);
+                        if (workflowDocument == null) {
+                            throw new RuntimeException("Unable to retrieve document # " +
+                                    cashControlDetail.getReferenceFinancialDocumentNumber() +
+                                    " from document service getByDocumentHeaderId");
+                        }
+                    }
+
+                    cashControlDetail.getReferenceFinancialDocument().getDocumentHeader()
+                            .setWorkflowDocument(workflowDocument);
+                }
+            }
+        }
+    }
+
+    public CashControlDocument getCashControlDocument() {
+        return (CashControlDocument) getDocument();
+    }
+
+    public CashControlDetail getNewCashControlDetail() {
+        if (newCashControlDetail == null) {
+            newCashControlDetail = new CashControlDetail();
+        }
+        return newCashControlDetail;
+    }
+
+    public void setNewCashControlDetail(final CashControlDetail newCashControlDetail) {
+        this.newCashControlDetail = newCashControlDetail;
+    }
+
+    public String getProcessingChartOfAccCodeAndOrgCode() {
+        return getCashControlDocument().getAccountsReceivableDocumentHeader()
+                .getProcessingChartOfAccCodeAndOrgCode();
+    }
+
+    public void setProcessingChartOfAccCodeAndOrgCode(final String processingChartOfAccCodeAndOrgCode) {
+        this.processingChartOfAccCodeAndOrgCode = processingChartOfAccCodeAndOrgCode;
+    }
+
+    public boolean isCashPaymentMediumSelected() {
+        return cashPaymentMediumSelected;
+    }
+
+    public void setCashPaymentMediumSelected(final boolean cashPaymentMediumSelected) {
+        this.cashPaymentMediumSelected = cashPaymentMediumSelected;
+    }
+
+    public FormFile getDetailImportFile() {
```

Comment:
A note should be added here on these functions and the new variable detailImportFile that they are part of the back port